libs/application/ui-shell/src/components/FormExternalDataProvider.tsx

```diff
@@ -51,6 +51,18 @@ const ItemHeader: React.FC<{ title: StaticText; subTitle?: StaticText }> = ({
   )
 }
 
+const isTranslationObject = (reason?: StaticText) => {
+  if (typeof reason !== 'object') {
+    return false
+  }
+
+  const hasId = reason.id !== undefined
+  const hasDescription = reason.description !== undefined
+  const hasDefaultMessage = reason.defaultMessage !== undefined
```

Comment:
Is the `id` check not enough? We have considered adding a preprocessor which removes description (and even defaultMessage) at build time to make our bundles smaller.